File: Java/Stack.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package src;

/**
 *
 * @author thwilliams
 */
public class Stack {
    //based off linked list implementation
    private static MyLinkedList l;
    public Stack(String input){
       this.l=new MyLinkedList(input);
    }
    public void push(String input){
        l.append(input);
    }
    public String pop(){
        String s=l.getLast();
        l.removeLast();
        return s;
    }
    public String toString(){
        //when printed out the stack grows downwards and shrinks upwards
        return l.toString();
    }
    public String peek(){
        return l.getLast();
    }
    public int size(){
        return l.size();
    }
    public boolean isEmpty(){
        return l.isEmpty();
    }
}

File: Python/stack.py
class Stack:
    def __init__(self, n=None):
        self.l=[n]
    def push(self, n):
        self.l.append(n)
    def pop(self):
        if(len(self.l)==0):
            print("Stack is empty")
            return 0
        temp=self.l[-1]
        self.l=self.l[:-1]
        return temp
    def peek(self):
        return self.l[-1]
    def __str__(self):
        return str(self.l)
    def size(self):
        return len(self.l)
    def clear(self):
        self.l=[]
    def isEmpty(self):
        return self.size()==0
    def setL(self, l):
        self.l=l
        return self
def listToStack(l):
    return Stack().setL(l)
if __name__=='__main__':
    stax=listToStack([1, 2, 3])
    stax.push(4)
    print(stax)